src/adminUtils.js

```diff
@@ -0,0 +1,199 @@
+  // Public functions
+  access(message, params) {
+    if (!params.length > 0) {
+      message.reply("Not enough details to show access");
+      return;
+    }
+
+    const username = params[0];
+    const discord_user = this._get_discord_user(message, username);
+    if (discord_user && this._admins()[discord_user.user.id]) {
+      this._print_access(message, username, discord_user.user.id);
+    } else {
+      message.reply(`${username} does not presently have any admin permissions`)
+    }
+  }
+
+  add(message, params) {
+    if (!params.length > 0) {
+      return;
+    }
+
+    const prefix = params[0];
+    if (!prefix.match(/^[a-z0-9_]+$/)) {
+      message.reply(`${prefix} is a bad short name`);
+    }
+
+    if (message.attachments.first()) {
+      // Only check the first attachment
+      const a = message.attachments.first();
+      const filename = `./Uploads/${prefix}--${a.filename}`;
+
+      log.debug(`Writing attachment to file: ${filename}`);
+      request(a.url).pipe(fs.createWriteStream(filename));
+
+      files[prefix] = filename;
+      message.reply(`!${prefix} is now available`);
+    } else {
+      message.reply(`You need to attach a file`);
+    }
+  }
+
+  grant(message, params) {
+    if (!params.length > 0) {
+      message.reply("Not enough details to add access");
+      return;
+    }
+
+    const username = params[0];
+    let access = params[1];
+    const discord_user = this._get_discord_user(message, username);
+    const adminList = this._admins();
+    if (discord_user && access) {
+      log.debug(`Updating: ${username} with ${access}`);
+      access = access.split(',').map(operation => operation.trim());
+      console.log(access);
+      const userId = discord_user.user.id;
+
+      if (!(userId in adminList)) {
+        log.debug("New user")
+        adminList[userId] = {'access': access, 'immune': false};
+      } else {
+        log.debug("Additional permissions")
+        adminList[userId]['access'] = [...access, ...adminList[userId]['access']]
+          .filter((element, pos, arr) => {
+            // Unique filter
+            return arr.indexOf(element) == pos;
+          });
+      }
+
+      this._saveConfig('adminList', adminList);
+      this._print_access(message, username, userId);
+    }
+  }
+
+  remove(message, params) {
+    if (!params.length > 0 || !(Object.keys(files).indexOf(params[0]) > -1)) {
+      log.debug(`Valid removal not found for: ${params}`)
+      return;
+    }
+
+    const clipName = params[0];
+    log.debug(`Deleting: ${files[clipName]}`);
+
+    fs.unlink(files[clipName], err => {
+      if (err) {
+        log.debug(err);
+        return;
+      }
+
+      delete files[clipName];
+      log.debug("Deleted successfully");
+      message.reply(`${clipName} removed`);
+    });
+  }
+
+  revoke(message, params) {
+    if (!params.length > 0) {
+      message.reply("Not enough details to remove access");
+      return;
+    }
+
+    const username = params[0];
```

Comment:
This code is repeated pretty often. Prime candidate for putting into a common function.